src/assets/scripts/client/airline/buildFlightNumber.js

```diff
@@ -0,0 +1,93 @@
+import { choose } from '../utilities/generalUtilities';
+import {
+    DEFAULT_CALLSIGN_FORMAT,
+    CALLSIGN_RANDOM_DIGIT_CHARACTER,
+    CALLSIGN_RANDOM_LETTER_CHARACTER
+} from '../constants/airlineConstants';
+
+const ALPHA = 'abcdefghijklmnopqrstuvwxyz';
+const NUMERIC = '0123456789';
+const defaultCallsignFormats = [DEFAULT_CALLSIGN_FORMAT];
+
+/**
+ * This picks a random number. If it is the first value within the callsign (ie. i === 0), then it picks a
+ * number between 1 and 9. Otherwise, it picks a number between 0 and 9.
+ *
+ * @function _generateRandomDigit
+ * @param i {number}
+ * @return NUMERIC {string}
+*/
+function _generateRandomDigit(i) {
+    if (i === 0) {
+        return choose(NUMERIC.substr(1));
+    }
+
+    return choose(NUMERIC);
+}
+
+/**
+ * This function picks a random letter and returns it.
+ *
+ * @function _generateRandomLetter
+ * @return ALPHA {string}
+*/
+function _generateRandomLetter() {
+    return choose(ALPHA);
+}
+
+/**
+ * This function takes a list of callsignFormats, validates them (ie. ensures that no callsignFormat has a leading 0)
+ * then returns a list of validated callsignFormats or the default callsignFormat if there are no valid callsignFormats
```

Comment:
Not worth changing, but just a note: `This function ...` doesn't really add much to the description, so to keep it concise, just come right out with it: `Validates and returns an array of validated cal...` or `Returns an array of validated cal...` etc. Shorter and clearer means faster and easier to read and understand 😄